DOC Add note for videos & improve syncing instructions (#20022)
scikit-learn/scikit-learn: doc/developers/contributing.rst

  `Transcript
  <https://github.com/data-umbrella/data-umbrella-scikit-learn-sprint/blob/master/3_transcript_ACM_video_vol2.md>`__

.. note::
  In January 2021, the default branch name changed from ``master`` to ``main``

>   for scikit-learn to remove any unnecessary references to slavery and

Reviewer: ```suggestion
  for scikit-learn GitHub repository, to remove any unnecessary references to slavery and
```

>   for scikit-learn to remove any unnecessary references to slavery and
>   replace them with more inclusive terms. These videos were created prior

Reviewer 2: ```suggestion
  for scikit-learn to use more inclusive terms. These videos were created 
  prior
```

  to the renaming of the branch. For contributors who are viewing these

>   videos to set up their working environment and submitting a PR, ``master``

Reviewer: ```suggestion
  videos to set up their working environment and submitting a pull-request, ``master``
```

Reviewer: Actually, we already introduce PR earlier.
